packages/host/tests/integration/components/card-basics-test.gts

```diff
@@ -2100,6 +2101,59 @@ module('Integration | card-basics', function (hooks) {
       assert.dom('[data-test-customer-billAmount]').containsText('100');
     });
 
+    test('re-renders a card with polymorphic primitive field', async function (assert) {
+      class SpecialStringAEmbedded extends Component<typeof SpecialStringA> {
+        <template>
+          <span data-test-polymorphic='special-string-a'>{{@model}}</span>
+        </template>
+      }
+      class SpecialStringA extends StringField {
+        static displayName = 'SpecialStringA';
+        static embedded = SpecialStringAEmbedded;
+      }
+      class SpecialStringBEmbedded extends Component<typeof SpecialStringB> {
+        <template>
+          <span data-test-polymorphic='special-string-b'>{{@model}}</span>
+        </template>
+      }
+      class SpecialStringB extends StringField {
+        static displayName = 'SpecialStringB';
+        static embedded = SpecialStringBEmbedded;
+      }
+      class TestCard extends CardDef {
+        static displayName = 'TestCard';
+        @field specialField = contains(StringField);
+        static isolated = class Isolated extends Component<typeof TestCard> {
+          setSubclass = () => {
+            this.args.model[fields] = { specialField: SpecialStringB };
+          };
```

Comment:
here we made a button to programmatically set a different polymorphic primitive field on our instance.